app/javascript/components/project-submissions/components/submissions-list.js

```diff
@@ -5,20 +5,58 @@ import FlipMove from 'react-flip-move';
 import Submission from './submission';
 import ProjectSubmissionContext from '../ProjectSubmissionContext';
 
-const noop = () => {};
+const noop = () => { };
 
 const SubmissionsList = ({ submissions, handleDelete, onFlag, handleUpdate, isDashboardView, handleLikeToggle, userSubmission }) => {
   const { allSubmissionsPath, legacySubmissionsUrl } = useContext(ProjectSubmissionContext);
   const hasSubmissions = submissions.length > 0;
-  
-  // if (userSubmission) {
-  //   submissions = submissions.filter((submission) => submission.user_id !== userSubmission.user_id);
-  // }
 
-  return (
-    <div>
+  console.log(isDashboardView);
+
+  if (isDashboardView) {
+    return (
+      <div>
+        { userSubmission
```

Comment:
I think this component is getting a bit too complicated by having to handle the lesson and dashboard submission lists. It will be very difficult to test when it comes to that.

Whats your thoughts on creating a user-submissions-list component that will just handle the dashboard list?

Reply:
Yeah that was going to be my first refactor. To be honest, I’m not sure that removing the ReactFlip even resolved the issue so we might be able to keep this the way it is. I didn’t resolve the duplication issue until I changed the update method itself.

Reply:
@KevinMulhern Confirmed React.Flip wasn't the problem, so I'm adding it back so this component doesn't need to be changed :)